app/components/windows/FFZSettings.vue.ts

```diff
@@ -0,0 +1,43 @@
+import Vue from 'vue';
+import { Component } from 'vue-property-decorator';
+import { Inject } from '../../util/injector';
+import ModalLayout from '../ModalLayout.vue';
+import { CustomizationService } from 'services/customization';
+import { WindowsService } from 'services/windows';
+
+@Component({
+  components: {
+    ModalLayout
+  }
+})
+export default class FFZSettings extends Vue {
+  @Inject() customizationService: CustomizationService;
+  @Inject() windowsService: WindowsService;
+
+  $refs: {
+    ffzSettings: Electron.WebviewTag
+  }
+
+  nightMode: Boolean = false;
+
+  mounted() {
+    const webview = this.$refs.ffzSettings;
+    const settings = this.customizationService.getSettings();
+
+    this.nightMode = settings.nightMode;
+
+    webview.addEventListener('dom-ready', () => {
+      webview.setZoomFactor(settings.chatZoomFactor);
+
+      webview.executeJavaScript(`
+        var ffzscript1 = document.createElement('script');
+        ffzscript1.setAttribute('src','https://cdn.frankerfacez.com/script/script.min.js');
+        document.head.appendChild(ffzscript1);
+      `, true);
+    })
+  }
+
+  get popoutURL() {
+    return `https://www.twitch.tv/popout/frankerfacez/chat?ffz-settings${this.nightMode ? '&darkpopout' : ''}`
```

Comment:
So this is loading the `frankerfacez` popout chat, and then the injected javascript just replaces the content of the page with your settings?  Can this be any `twitch.tv` URL?  Just curious.

Reply:
That can technically be any Twitch URL, yes. When FFZ loads, it checks for a `ffz-settings` query parameter and, if it's present, it displays its settings UI in exclusive mode. I use the chat pop-out URL specifically because it's a fairly lightweight page and because it initializes chat data, so certain parts of the FFZ settings can populate correctly.

As an example, the Badge Visibility settings are dependent on chat having been loaded to populate the list of available Twitch badges.